ui/src/main/resources/PhenoTips/MatchContact.xml

```diff
@@ -54,7 +54,10 @@
       'groups'   : "#if($options.groups)${groups}#end",
 
       'subject'  : "[$!{platformName}] $!{options.subject}",
-      'message'  : "$!{options.message}"
+      'message'  : "$!{options.message}",
+
+      'patientURL': $xwiki.getDocument($services.patients.getPatientById("$!{request.patient}").getDocument()).getExternalURL(),
+      'matchURL'  : $xwiki.getDocument($services.patients.getPatientById("$!{request.matchId}").getDocument()).getExternalURL()
```

Comment:
It's not pretty, but you could do:

    $xwiki.getDocument($services.patients.getPatientById("$!{request.patient}").getDocument()).getExternalURL().replace($request.patient, $request.matchId)